c/validation/CMakeLists.txt

```diff
@@ -0,0 +1,58 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+cmake_minimum_required(VERSION 3.18)
+include(CMakePackageConfigHelpers)
+project(adbc_validation LANGUAGES CXX)
+
+find_package(GTest REQUIRED)
+get_filename_component(REPOSITORY_ROOT "../../" ABSOLUTE)
+
+add_library(adbc_validation STATIC adbc_validation.cc adbc_validation_util.cc)
+target_compile_features(adbc_validation PRIVATE cxx_std_17)
+target_include_directories(adbc_validation SYSTEM PRIVATE "${REPOSITORY_ROOT}"
+                                                          "${REPOSITORY_ROOT}/c/vendor/")
+target_link_libraries(adbc_validation PUBLIC nanoarrow GTest::gtest GTest::gmock)
+
+set_target_properties(adbc_validation
+                      PROPERTIES PUBLIC_HEADER "adbc_validation.h;adbc_validation_util.h")
+
+install(TARGETS adbc_validation
+        EXPORT AdbcValidationTargets
+        LIBRARY DESTINATION lib
+        PUBLIC_HEADER DESTINATION include/adbc_validation/)
+
+write_basic_package_version_file(
+  "${CMAKE_CURRENT_BINARY_DIR}/AdbcValidation/AdbcValidationConfigVersion.cmake"
+  VERSION main
```

Comment:
Can we use ADBC's version here? Or we can omit this by specifying ADBC's version in `project()`:

https://cmake.org/cmake/help/latest/module/CMakePackageConfigHelpers.html#generating-a-package-version-file

> If no VERSION is given, the [PROJECT_VERSION](https://cmake.org/cmake/help/latest/variable/PROJECT_VERSION.html#variable:PROJECT_VERSION) variable is used. If this hasn't been set, it errors out.

Reply:
Updated here to use GNUInstallDirs. I split the version define into a new file containing only the version; I'll update the other PR once this lands.